elasticsearch-model/lib/elasticsearch/model/adapters/multiple.rb

```diff
@@ -106,16 +106,16 @@ def __ids_by_type
           # @api private
           #
           def __type_for_hit(hit)
-            @@__types ||= {}
 
             key = "#{hit[:_index]}::#{hit[:_type]}" if hit[:_type] && hit[:_type] != '_doc'
             key = hit[:_index] unless key
 
-            @@__types[key] ||= begin
```

Comment:
I put Benchmark.measure around this code and contrary to my expectations, the original hash method was WAY faster.  We will have to re-think this change, maybe the method of storing the class->index_name mapping can be improved to mitigate #838 with performance.